src/scenechange/mod.rs

```diff
@@ -354,67 +393,64 @@ impl<T: Pixel> SceneChangeDetector<T> {
+    let mut mv_inter_cost = 0.0;
+    let mut hg_inter_cost = 0.0;
+    crate::rayon::scope(|s| {
+      s.spawn(|_| {
         let intra_costs = estimate_intra_costs(
           &*frame2,
           self.bit_depth,
           self.cpu_feature_level,
         );
-        intra_costs.iter().map(|&cost| cost as u64).sum::<u64>() as f64
+        intra_cost = intra_costs.iter().map(|&cost| cost as u64).sum::<u64>()
+          as f64
           / intra_costs.len() as f64
-      },
-      move || {
-        let inter_costs = if self.speed_mode == SceneDetectionSpeed::Medium {
-          estimate_inter_costs(
-            frame2_ref2,
-            frame1,
-            self.bit_depth,
-            self.encoder_config,
-            self.sequence.clone(),
-          )
-        } else {
-          estimate_inter_costs_histogram_blocks(frame2_ref2, frame1)
-        };
-
-        inter_costs.iter().map(|&cost| cost as u64).sum::<u64>() as f64
-          / inter_costs.len() as f64
-      },
-    );
+      });
+      s.spawn(|_| {
+        let inter_costs = estimate_inter_costs(
+          frame2_ref2,
+          frame1,
+          self.bit_depth,
+          self.encoder_config,
+          self.sequence.clone(),
+        );
 
-    // Sliding scale, more likely to choose a keyframe
-    // as we get farther from the last keyframe.
-    // Based on x264 scenecut code.
-    //
-    // `THRESH_MAX` determines how likely we are
+        mv_inter_cost =
+          inter_costs.iter().map(|&cost| cost as u64).sum::<u64>() as f64
+            / inter_costs.len() as f64
+      });
+      s.spawn(|_| {
+        let inter_costs =
+          estimate_inter_costs_histogram_blocks(frame2_ref3, frame1_ref2);
+
+        hg_inter_cost =
+          inter_costs.iter().map(|&cost| cost as u64).sum::<u64>() as f64
+            / inter_costs.len() as f64
+      });
+    });
+
+    // `BIAS` determines how likely we are
     // to choose a keyframe, between 0.0-1.0.
     // Higher values mean we are more likely to choose a keyframe.
-    // `0.833` was chosen based on trials using the new
+    // This value was chosen based on trials using the new
     // adaptive scenecut code.
-    const THRESH_MAX: f64 = 0.833;
-    const THRESH_MIN: f64 = 0.75;
-    let distance_from_keyframe = frameno - previous_keyframe;
-    let min_keyint = self.encoder_config.min_key_frame_interval;
-    let max_keyint = self.encoder_config.max_key_frame_interval;
-    debug_assert!(distance_from_keyframe >= min_keyint);
-    let bias = THRESH_MIN
-      + (THRESH_MAX - THRESH_MIN)
-        * (distance_from_keyframe - min_keyint) as f64
-        / (max_keyint - min_keyint) as f64;
-
-    // Adaptive threshold for medium version, static thresholf for the slow one
-    let threshold = if self.speed_mode == SceneDetectionSpeed::Medium {
-      intra_cost * (1.0 - bias)
-    } else {
-      self.threshold as f64
-    };
+    const BIAS: f64 = 0.7;
+    let threshold = intra_cost * (1.0 - BIAS);
```

Comment:
Did you find the ramp when we approach max keyframe interval to not be useful?

Reply:
OK. Maybe we can revisit it later. It'll definitely create false positives in terms of "real" scenecuts, so the only way to really test it is with AWCY numbers rather than any sort of "keyframe ground truth".